MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp

```diff
@@ -289,5 +368,46 @@ std::string QtReflMainView::getTransferMethod() const {
   return ui.comboTransferMethod->currentText().toStdString();
 }
 
+/**
+Set the list of tables the user is offered to open
+@param tables : the names of the tables in the ADS
+*/
+void QtReflMainView::setTableList(const std::set<std::string> &tables) {
+  ui.menuOpenTable->clear();
```

Comment:
If you where to push `Command`s at this level, then you would simply need to enumerate over those `Commands` in the view